react/src/base/inputs/SprkCheckbox/SprkCheckboxGroup/SprkCheckboxGroup.js

```diff
@@ -11,51 +10,61 @@ const SprkCheckboxGroup = (props) => {
     idString,
     additionalClasses,
     analyticsString,
+    ariaDescribedBy,
+    ...rest
   } = props;
 
-  let errorId = null;
-  let helperId = null;
-  let hasErrorContainer = false;
-  let hasHelperText = false;
-
+  let errorId;
+  let helperId;
+  let inputAriaDescribedByArray = [];
+  if (ariaDescribedBy) {
+    inputAriaDescribedByArray = ariaDescribedBy.split(' ');
+  }
   const childrenArray = React.Children.toArray(children);
-  // Map through children, assign SprkErrorContainer with id/uniqueId
-  // Assign errorId which will determine the ariaDescribedBy of SprkCheckboxItem
+  // Get the errorId and helperId.
+  // If they don't exist in the ariaDescribedBy, add them.
   const elementsToProcess = childrenArray.map((element) => {
-    if (element.type.name === 'SprkErrorContainer') {
-      hasErrorContainer = true;
-      errorId = element.props.id || uniqueId('sprk-error-container-');
-      return React.cloneElement(element, { id: errorId });
+    if (
+      element.type.name === 'SprkFieldError' ||
+      element.type.name === 'SprkErrorContainer'
+    ) {
+      errorId = element.props.id;
+      if (!inputAriaDescribedByArray.includes(errorId)) {
+        inputAriaDescribedByArray.push(errorId);
+      }
+      return React.cloneElement(element);
```

Comment:
This makes sense, it grabs the ids of the elements and makes sure they're in the `ariaDescribedBy` of the element. This is what creates the relationship.

One thing is that these AriaDescribedBy is being added to the **container element**. 

Shouldn't the relationships be forged with the input? The div surrounding it doesn't really need to know anything about what it's described by.

From my understanding this is creating a relationship between the **container** and the Field Error, Error Container, and Helper Text.

But what makes more sense to me is that the **input** is described by:
- Field Error
- Error Container
- And the helper text